File: app/src/main/java/org/dicio/dicio_android/settings/skill/SkillsAdapter.java
package org.dicio.dicio_android.settings.skill;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;

import org.dicio.dicio_android.skills.SkillHandler;
import org.dicio.skill.SkillInfo;

import java.util.List;

public class SkillsAdapter extends RecyclerView.Adapter<SkillItemHolder> {

    final Fragment fragment;
    @NonNull final LayoutInflater inflater;
    final List<SkillInfo> skillInfoList;

    public SkillsAdapter(final Fragment fragment,
                         @NonNull final LayoutInflater inflater,
                         final List<SkillInfo> skillInfoList) {
        this.fragment = fragment;
        this.inflater = inflater;
        this.skillInfoList = skillInfoList;
    }

    @NonNull
    @Override
    public SkillItemHolder onCreateViewHolder(@NonNull final ViewGroup parent, final int viewType) {
        return new SkillItemHolder(inflater, parent);
    }

    @Override
    public void onBindViewHolder(@NonNull final SkillItemHolder holder, final int position) {
        holder.bind(fragment, skillInfoList.get(position));
    }

    @Override
    public void onViewRecycled(@NonNull final SkillItemHolder holder) {
        holder.unbind(fragment);
    }

    @Override
    public int getItemCount() {
        return skillInfoList.size();
    }
}

File: app/src/main/kotlin/org/stypox/dicio/settings/skill/SkillsAdapter.kt
package org.stypox.dicio.settings.skill

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView
import org.dicio.skill.SkillInfo

class SkillsAdapter(
    private val fragment: Fragment,
    private val inflater: LayoutInflater,
    private val skillInfoList: List<SkillInfo>
) : RecyclerView.Adapter<SkillItemHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SkillItemHolder {
        return SkillItemHolder(inflater, parent)
    }

    override fun onBindViewHolder(holder: SkillItemHolder, position: Int) {
        holder.bind(fragment, skillInfoList[position])
    }

    override fun onViewRecycled(holder: SkillItemHolder) {
        holder.unbind(fragment)
    }

    override fun getItemCount(): Int {
        return skillInfoList.size
    }
}